aws/resource_aws_eks_node_group.go

```diff
@@ -323,29 +324,29 @@ func resourceAwsEksNodeGroupCreate(d *schema.ResourceData, meta interface{}) err
 	_, err := conn.CreateNodegroup(input)
 
 	if err != nil {
-		return fmt.Errorf("error creating EKS Node Group (%s): %w", id, err)
+		return diag.Errorf("error creating EKS Node Group (%s)", id, err)
 	}
 
 	d.SetId(id)
 
-	_, err = waiter.NodegroupCreated(conn, clusterName, nodeGroupName, d.Timeout(schema.TimeoutCreate))
+	_, err = waiter.NodegroupCreated(ctx, conn, clusterName, nodeGroupName, d.Timeout(schema.TimeoutCreate))
 
 	if err != nil {
-		return fmt.Errorf("error waiting for EKS Node Groupe (%s) to create: %w", d.Id(), err)
+		return diag.Errorf("error waiting for EKS Node Groupe (%s) to create", d.Id(), err)
```

Comment:
Still need to pass `error`, just can't use `%w`.

```suggestion
		return diag.Errorf("error waiting for EKS Node Group (%s) to create: %s", d.Id(), err)
```